### PR TITLE
Allow preload to accept a const string.

### LINKED ISSUE #6798: Preload function would have to accept a  constant string parameter as input.
**Preload function would have to accept a  constant string parameter as input.**

```
const WARM_UP_STATE = "res://scenes/states/warm_up/warm_up.tscn"

func _go():
    current_state = preload(WARM_UP_STATE).instance()
    level.add_child(current_state)

```

> res://scenes/levels/level_01/src/node2d_state_manager.gd:23  - Parse Error: expected string constant as 'preload' argument.

### PR DESCRIPTION
**Pull request by hpvb**
Allow preload to accept a const string.

In preload() parsing this code will lookup the identifier in the local
constant database. If the identifier corresponds to a string constant
it is used as the path for preload().

Currently this does not work for global constants, only constants
declared in the same class as the preload is happening. We can implement
a full fix too. Maybe we can use this PR to discuss the possibilities.

This (partially) fixes #6798

### CODE REVIEW

**Comment by hpvb** [comment]
This may actually need a little more work as constants not defined in the current class currently can't be resolved.

**Comment by reduz** [comment]
At much it should work for current class only

File: modules/gdscript/gd_parser.cpp [hunk 1]
```diff
@@ -386,16 +386,35 @@ GDParser::Node* GDParser::_parse_expression(Node *p_parent,bool p_static,bool p_
...
 					}
 				}
+
+				if (found_constant && cn->value.get_type() == Variant::STRING) {
+					valid = true;
+					path = (String) cn->value;
+				}
 			}
 			if (!valid) {
 				_set_error("expected string constant as 'preload' argument.");
```
Comment (bojidar-bg) [inline modules/gdscript/gd_parser.cpp]: You forgot to update the message here

File: modules/gdscript/gd_parser.cpp [hunk 2]
```diff
@@ -1123,13 +1142,13 @@ GDParser::Node* GDParser::_parse_expression(Node *p_parent,bool p_static,bool p_
...
 
-				
+
 				case OperatorNode::OP_IN: priority=10; break;
-				
+
 				case OperatorNode::OP_NOT: priority=11; unary=true; break;
 				case OperatorNode::OP_AND: priority=12; break;
 				case OperatorNode::OP_OR: priority=13; break;
-
```
Comment (bojidar-bg) [inline modules/gdscript/gd_parser.cpp]: All of those changes shouldn't have happened, since they are unrelated, but let's say it might pass....

**Comment by hpvb** [comment]
@reduz  wouldn't it make sense that it should also work for constants that came in through autoloaded classes?

File: modules/gdscript/gd_parser.cpp (hunk 2 above)
Comment (hpvb) [inline modules/gdscript/gd_parser.cpp]: I completely didn't notice that. I'll clean up the PR, sorry.

File: modules/gdscript/gd_parser.cpp (hunk 1 above)
Comment (hpvb) [inline modules/gdscript/gd_parser.cpp]: Do you have a suggestion? I figured 'string constant' was still pretty accurate.

**Review by akien-mga — APPROVED** [review]

**Pull request merged**